apps/etterlatte-samordning-vedtak/src/main/kotlin/no/nav/etterlatte/samordning/vedtak/SamordningVedtakRoute.kt

```diff
@@ -22,14 +22,25 @@ fun Route.samordningVedtakRoute(samordningVedtakService: SamordningVedtakService
         get("{vedtakId}") {
             val vedtakId = requireNotNull(call.parameters["vedtakId"]).toLong()
 
+            val tpnummer =
+                call.request.headers["tpnr"]
+                    ?: return@get call.respond(HttpStatusCode.BadRequest, "tpnr ikke angitt")
+
             val samordningVedtakDto =
                 try {
                     samordningVedtakService.hentVedtak(
                         vedtakId = vedtakId,
+                        tpnr = tpnummer,
                         organisasjonsnummer = call.orgNummer,
                     )
                 } catch (e: VedtakFeilSakstypeException) {
                     call.respond(HttpStatusCode.Unauthorized, "Ikke tilgang til sakstype")
+                } catch (e: TjenestepensjonManglendeTilgangException) {
```

Comment:
Supert med spesifikke exceptions, særleg når det går ut mot tredjepart som her.

Hadde vore kult om vi kan få til ein generalisert versjon etter kvart, utan at eg ser heilt for meg akkurat korleis.

Reply:
Ja, mon tro om @oyvindsh har fått sett noe på det